SudoScript.Core/Solver.cs

```diff
@@ -71,14 +72,96 @@ private static bool SolveRec(Board board, [NotNullWhen(true)] out Board? solvedB
         return false;
     }
 
+    /// <summary>
+    /// Finds a number of solutions to a given board.
+    /// </summary>
+    /// <param name="board"></param>
+    /// <param name="random">Whether the solved boards should be randomized.</param>
+    /// <param name="solutionCount">Maximum number of solutions that should be in the returned list. Zero means unlimited.</param>
+    /// <returns></returns>
+    public static List<Board>? FindSolutions(Board board, int limit = 0, bool random = false)
```

Comment:
I would reccommend against using magic numbers as you do with 0 here if at all possible. If you instead set limit to int.MaxValue you will also ensure to not generate more solutions than can be stored in a list.

However, i dont get why we are even storing these in a list at all. Seems like it would both be simpler and safer to store them directly in an IEnumerable. Then the user can just say .Take themselves.

Reply:
Upon looking a bit further, since you are creating these in a hashset returning an IEnumerable might not be the best idea anyways. If there is a way to do it that would be good. But it would make duplicate detection a bigger problem. If you can figure out how to do it that would be really nice. Otherwise you can go ahead with a list. (You could also consider returning the hashset directly which might honestly be my prefered option.

Still change the 0 to be int.MaxValue though.

Reply:
Adressed with [73a7bd8].
I will look into returning in a different way as I test these following days. This pull request stands for now.